### PR TITLE
Prevent tagsinput select element to be submitted without a value when they are required

### LINKED ISSUE #11274: At times private messages are sent to random users
We have discussed this here https://social.trom.tf/display/dbc8dc44-1462-13a9-5919-bb3918828493

Basically I was sending a private message to someone who was not in my contacts list, and the message landed on a random user's "inbox". Others have reported it too. 

`This is Friendica, version 2021.09 that is running at the web location https://social.trom.tf. The database version is 1434/1434, the post update version is 1427/1427.`

### PR DESCRIPTION
**Pull request by MrPetovan**
Fixes #11274 

Additionally, we now perform the same validity check when sending a private message than we do when compiling the list of valid recipients for display in the new message page.

### CODE REVIEW

**Comment by annando** [comment]
@MrPetovan a check is failing.

**Comment by MrPetovan** [comment]
Test is fixed, since we're now checking the actual validity of the recipient when we create a private message, the test fixtures required an additional field set, `contact.notify` that we check on.

File: src/Core/ACL.php [hunk 1]
```diff
@@ -62,25 +62,7 @@ public static function getMessageContactSelectHTML(int $selected = null)
...
-		];
-
-		$contacts = Contact::selectToArray(
-			['id', 'name', 'addr', 'micro'],
-			DBA::mergeConditions($condition, ["`notify` != ''"])
-		);
-
-		$arr = ['contact' => $contacts, 'entry' => $o];
-
-		Hook::callAll(DI::args()->getModuleName() . '_pre_recipient', $arr);
```
Comment (annando) [inline src/Core/ACL.php:83]: Have you dropped that hook by intention?

File: src/Model/Mail.php [hunk 2]
```diff
@@ -170,7 +175,7 @@ public static function send($recipient = 0, $body = '', $subject = '', $replyto
 			$recip_host = substr($contact['url'], strpos($contact['url'], '://') + 3);
 			$recip_host = substr($recip_host, 0, strpos($recip_host, '/'));
 
-			$recip_handle = (($contact['addr']) ? $contact['addr'] : $contact['nick'] . '@' . $recip_host);
+			$recip_handle = $contact['addr'] ?: $contact['nick'] . '@' . $recip_host;
```
Comment (annando) [inline src/Model/Mail.php]: This is some ancient code. We now always have got the contact's address in that field when the remote system does support this format. So the code here can be made slightly simpler.

BTW: The same is valid for the `$sender_handle`: This field is always filled and doesn't need to be constructed.

File: src/Core/ACL.php (hunk 1 above)
Comment (MrPetovan) [reply src/Core/ACL.php:83]: Yes, this hook isn't document, it isn't used in addons, and even if it was used in a custom addon the potentially modified values in `$arr` aren't used after the hook call, which makes it both trying to do 3 things at once (change viable message target list, change the HTML output itself and be module-specific) and failing at two of them.

File: src/Model/Mail.php (hunk 2 above)
Comment (MrPetovan) [reply src/Model/Mail.php]: Thanks for the heads-up!

Comment (MrPetovan) [reply src/Model/Mail.php]: Done. It was readily available indeed.

**Pull request merged**